site/app/controllers/HomePageController.php

```diff
@@ -95,6 +96,20 @@ public function changeUserName() {
             else {
                 $this->core->addErrorMessage("Preferred names must not exceed 30 chars.  Letters, spaces, hyphens, apostrophes, periods, parentheses, and backquotes permitted.");
             }
+
+            // If we received an image file attempt to save it
+            if ($_FILES['user_image']['tmp_name'] !== '') {
+                $meta = explode('.', $_FILES['user_image']['name']);
+                $file_name = $meta[0];
+                $extension = $meta[1];
+
+                // Save image for user
+                $result = $user->setDisplayImage($file_name, $extension, $_FILES['user_image']['tmp_name']);
```

Comment:
Please move the DB and file system logic out of the model. The models should be just containers for information that the controller uses to do whatever. Adding the functionality into the model makes testing them that much more difficult.

Reply:
I'm going to shelve this until a later PR.  See #5587.